web/app/scripts/state/recordstate-service.js

```diff
@@ -20,6 +27,8 @@
          */
         function init() {
           selected = null;
+          gettingSelected = false;
+          gettingOptions = false;
```

Comment:
They should definitely be extended to the other scripts - the main purpose is just to reduce the number of outgoing requests.